NullPointerException parsing 10302018.NASDAQ_ITCH50

I am trying to parse the uncompressed file 10302018.NASDAQ_ITCH50 (originally from ftp://emi.nasdaq.com/ITCH/10302018.NASDAQ_ITCH50.gz). It fails consistently with the below stack:

Exception in thread "main" java.lang.NullPointerException
        at ParseDS.getFields(ParseDS.java:48)
        at Parsers.messageIn(Parsers.java:47)
        at Parse.parse(Parse.java:61)
        at Parse.main(Parse.java:93)

The tail of the output file shows the below lines:

[D, 1517, 0, 34790347615211, 11491833]
[A, 8592, 0, 34790347635580, 41345596, B, 100, XLP, 54.96]
[A, 8592, 0, 34790347640759, 41345600, S, 100, XLP, 55.03]
[D, 8265, 0, 34790347645719, 41339864]
[U, 7379, 0, 34790347647139, 41344468, 41345608, 14, 104.42]
[A, 3599, 0, 34790347666980, 30131022, B, 100, HAS, 92.23]
[A, 5518, 0, 34790347672809, 30821291, S, 100, NKE, 73.41]
[A, 8265, 0, 34790347706962, 41345616, B, 300, VOOG, 140.6]
[A, 3599, 0, 34790347707980, 30131026, B, 100, HAS, 92.23]
[D, 8216, 0, 34790347708330, 41345308]

I re-ran with -enableassertions and it says:

Exception in thread "main" java.lang.AssertionError: File type missing: J

The YAML file does not seem to have an entry for J. Neither does the spec (https://www.nasdaqtrader.com/content/technicalsupport/specifications/dataproducts/NQTVITCHspecification.pdf). Is this corrupt data?

Since it is a NullPointerException I thought I would report this. Hopefully you can reproduce this and help with a fix. Many thanks in advance!

--- a/itch5.yaml
+++ b/itch5.yaml
@@ -58,6 +58,10 @@
     - IPOQuotRelTime               : [2,4]
     - IPOQuotRelQual               : [1,1]
     - IPOPrice                     : [3,4]
+    - AuctionCollarReferencePrice  : [3,4]
+    - UpperAuctionCollarPrice      : [3,4]
+    - LowerAuctionCollarPrice      : [3,4]
+    - AuctionCollarExtension       : [2,4] 
     messages:
       S:
         name: System Event Message
@@ -88,6 +92,18 @@
         - ETPFlag
         - ETPFactor
         - InverseIndicator
+      J:
+        name: LULD Auction Collar
+        fields:
+        - MessageType
+        - StockLocate
+        - TrackingNumber
+        - Timestamp
+        - Stock
+        - AuctionCollarReferencePrice
+        - UpperAuctionCollarPrice
+        - LowerAuctionCollarPrice
+        - AuctionCollarExtension 
       H:
         name: Stock Trading Action
         fields:

